firebase-sessions/src/test/kotlin/com/google/firebase/sessions/SessionLifecycleServiceTest.kt

```diff
@@ -0,0 +1,253 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.sessions
+
+import android.content.Context
+import android.content.Intent
+import android.os.Handler
+import android.os.Looper
+import android.os.Message
+import android.os.Messenger
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.filters.MediumTest
+import com.google.common.truth.Truth.assertThat
+import com.google.firebase.Firebase
+import com.google.firebase.FirebaseApp
+import com.google.firebase.FirebaseOptions
+import com.google.firebase.initialize
+import com.google.firebase.sessions.testing.FakeFirebaseApp
+import com.google.firebase.sessions.testing.FakeFirelogPublisher
+import com.google.firebase.sessions.testing.FakeSessionDatastore
+import java.time.Duration
+import java.util.concurrent.TimeoutException
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.Robolectric
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.Shadows.shadowOf
+import org.robolectric.android.controller.ServiceController
+import org.robolectric.annotation.LooperMode
+import org.robolectric.annotation.LooperMode.Mode.PAUSED
+import org.robolectric.shadows.ShadowSystemClock
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@MediumTest
+@LooperMode(PAUSED)
+@RunWith(RobolectricTestRunner::class)
+internal class SessionLifecycleServiceTest {
+
+  lateinit var service: ServiceController<SessionLifecycleService>
+  lateinit var firebaseApp: FirebaseApp
+
+  data class CallbackMessage(val code: Int, val sessionId: String?)
+
+  internal inner class TestCallbackHandler(looper: Looper = Looper.getMainLooper()) :
+    Handler(looper) {
+    val callbackMessages = ArrayList<CallbackMessage>()
+
+    override fun handleMessage(msg: Message) {
+      callbackMessages.add(CallbackMessage(msg.what, getSessionId(msg)))
+    }
+  }
+
+  @Before
+  fun setUp() {
+    val context = ApplicationProvider.getApplicationContext<Context>()
+    firebaseApp =
+      Firebase.initialize(
+        ApplicationProvider.getApplicationContext(),
+        FirebaseOptions.Builder()
+          .setApplicationId(FakeFirebaseApp.MOCK_APP_ID)
+          .setApiKey(FakeFirebaseApp.MOCK_API_KEY)
+          .setProjectId(FakeFirebaseApp.MOCK_PROJECT_ID)
+          .build()
+      )
+    service = createService()
+  }
+
+  @After
+  fun cleanUp() {
+    FirebaseApp.clearInstancesForTest()
+  }
+
+  @Test
+  @Throws(TimeoutException::class)
```

Comment:
Do we need all these throws annotations?

Reply:
Removed